core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -587,20 +514,175 @@ private void accumulateDemand(long n) {
             }
         }
 
+        void signal() {
+            if (executor == null) {
+                doSignal();
+            } else {
+                executor.execute(this::doSignal);
+            }
+        }
+
+        private void doSignal() {
+            final List<Object> signals = processor.signals();
+            final int endOffset = signals.size();
+            for (;;) {
+                if (cancelledOrAborted != null) {
+                    if (beginSignal()) {
+                        processor.unsubscribe(this, cancelledOrAborted);
+                        // Don't have to bring notifying back to 0 because it's completed.
```

Comment:
We can take this sentence out or change something like "Don't have to call endSignal() because~~"